cmd/convox/init_test.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/convox/rack/manifest"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInitReadProcfile(t *testing.T) {
+
+	procfileTests := []struct {
+		data     []byte
+		procfile Procfile
+	}{
+		{[]byte(rubyProcfile), Procfile{{"web", "bundle exec puma -C config/puma.rb"}}},
+		{[]byte(pythonProcfile), Procfile{{"web", "gunicorn gettingstarted.wsgi --log-file -"}}},
+		{[]byte(nodejsProcfile), Procfile{{"web", "node index.js"}}},
+	}
+
+	for _, pt := range procfileTests {
+		pf := ReadProcfile(pt.data)
+		assert.Equal(t, pt.procfile, pf)
+	}
+
+}
+
+func TestInitReadAppfile(t *testing.T) {
+
+	appfileTests := []struct {
+		data []byte
+		af   Appfile
+	}{
+		{[]byte(appWithNoAddonsNoEnv), Appfile{[]string{}, nil}},
+		{[]byte(appWithPostgres), Appfile{[]string{"heroku-postgresql"}, nil}},
+		{[]byte(appWithEnvAndPostgres), Appfile{[]string{"heroku-postgresql"}, map[string]EnvEntry{"DEBUG": {"1"}, "SECRET_TOKEN": {"secret"}}}},
+	}
+
+	for _, at := range appfileTests {
+		af, err := ReadAppfile(at.data)
+		assert.NoError(t, err)
+		assert.Equal(t, at.af, af)
+	}
+
+}
+
+func TestGenerateManifest(t *testing.T) {
+	manifestTests := []struct {
+		pf Procfile
+		af Appfile
+		r  Release
+		m  manifest.Manifest
+	}{
+		{ //
```

Comment:
I was using `//` as a beginning/end block for input and output for a given test.